azext_iot/tests/test_iot_pnp_int.py

```diff
@@ -0,0 +1,183 @@
+            model_newContent = model.replace(json.loads(model)['@id'], self.kwargs['model_id'])
+            model_newContent = model_newContent.replace('\n', '')
+
+            fo = open(self.kwargs['model'], "w+", encoding='utf-8')
+            fo.write(model_newContent)
+            fo.close()
+
+    # TODO: @anusapan - Maybe put a helper like this in the shared lib, when you create it?
+    def command_execute_assert(self, command, asserts):
+        from iotext_test_tools import capture_output
+
+        with capture_output() as buffer:
+            self.cmd(command, checks=None)
+            output = buffer.get_output()
+
+        for a in asserts:
+            assert a in output
+
+    def tearDown(self):
+        change_dir()
+        if exists(self.kwargs['interface-updated']):
+            os.remove(self.kwargs['interface-updated'])
+        if exists(self.kwargs['model-updated']):
+            os.remove(self.kwargs['model-updated'])
+        if exists(self.kwargs['interface']):
+            os.remove(self.kwargs['interface'])
+        if exists(self.kwargs['model']):
+            os.remove(self.kwargs['model'])
+
+    def test_interface_life_cycle(self):
+
+        # Error: missing repo-id or login
+        self.cmd('iot pnp interface create -e {endpoint} --def {interface}', expect_failure=True)
+
+        # Error: Invalid Interface definition file
+        self.cmd('iot pnp interface create -e {endpoint} -r {repo} --def interface', expect_failure=True)
+
+        # Error: wrong path of Interface definition
+        self.cmd('iot pnp interface create -e {endpoint} -r {repo} --def interface.json', expect_failure=True)
+
+        # Success: Create new Interface
+        self.cmd('iot pnp interface create -e {endpoint} -r {repo} --def {interface}', checks=self.is_empty())
+
+        # Checking the Interface list
+        self.cmd('iot pnp interface list -e {endpoint} -r {repo}',
+                 checks=[self.greater_than('length([*])', 0)])
+
+        # Get Interface
+        interface = self.cmd('iot pnp interface show -e {endpoint} -r {repo} -i {interface_id}').get_output_in_json()
+        assert json.dumps(interface)
+        assert interface['@id'] == self.kwargs['interface_id']
+        assert interface['displayName'] == 'MXChip1'
+        assert len(interface['contents']) > 0
+
+        # Success: Update Interface
+        interface = str(read_file_content(self.kwargs['interface']))
+        display_name = json.loads(interface)['displayName']
+        interface_newContent = interface.replace(display_name, '{}-Updated'.format(display_name))
+        interface_newContent = interface_newContent.replace('\n', '')
+        fo = open(self.kwargs['interface-updated'], "w+", encoding='utf-8')
+        fo.write(interface_newContent)
+        fo.close()
+        self.cmd('iot pnp interface update -e {endpoint} -r {repo} --def {interface-updated}', checks=self.is_empty())
+
+        # Todo: Publish Interface
```

Comment:
Please add a test for Interface Publish.

Reply:
Added the test